src/table/body-cell/styles.scss

```diff
@@ -44,24 +58,47 @@ $interactive-column-padding-inline-end: calc(#{$cell-horizontal-padding} + #{aws
   margin-inline-start: calc(-1 * (#{awsui.$space-m} + #{awsui.$space-xs}));
 }
 
-@mixin cell-offset($padding) {
-  padding-inline-start: $padding;
+@mixin cell-padding-inline-start($padding) {
+  > .body-cell-content {
+    padding-inline-start: $padding;
+  }
 
   @for $i from 0 through 9 {
-    &.expandable-level-#{$i} {
+    &.expandable-level-#{$i} > .body-cell-content {
       padding-inline-start: calc($padding + $i * (#{awsui.$space-m} + #{awsui.$space-xs}));
     }
   }
-  &.expandable-level-next {
+  &.expandable-level-next > .body-cell-content {
     padding-inline-start: calc($padding + 9 * (#{awsui.$space-m} + #{awsui.$space-xs}));
   }
 }
+@mixin cell-padding-inline-end($padding) {
+  > .body-cell-content {
+    padding-inline-end: $padding;
+  }
+}
+@mixin cell-padding-block($padding) {
+  > .body-cell-content {
+    padding-block: calc(#{$padding} - 1 * #{awsui.$border-divider-list-width});
+  }
+}
+@mixin cell-padding-block-start($padding) {
+  > .body-cell-content {
+    padding-block-start: calc(#{$padding} - 1 * #{awsui.$border-divider-list-width});
+  }
+}
+@mixin cell-padding-block-end($padding) {
+  > .body-cell-content {
+    padding-block-end: calc(#{$padding} - 1 * #{awsui.$border-divider-list-width});
+  }
+}
 
 .body-cell {
   box-sizing: border-box;
-  padding-block-start: $cell-vertical-padding;
-  padding-block-end: $cell-vertical-padding-w-border;
-  padding-inline-end: $cell-horizontal-padding;
+  @include cell-padding-inline-start($cell-horizontal-padding);
+  @include cell-padding-block-start($cell-vertical-padding);
+  @include cell-padding-block-end($cell-vertical-padding-w-border);
+  @include cell-padding-inline-end($cell-horizontal-padding);
```

Comment:
The mixins are not strictly necessary but they allow to make the code difference minimal and arguable improve readability by avoiding too much nesting. There is an opportunity to refactor the styles further by making body and body content styles less coupled.